cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -102,23 +102,119 @@ class column_wrapper {
+      begin, end, out, [](auto const& e) { return TargT{typename TargT::duration{e}}; });
+  }
+#endif
+
+  // Convert timestamps to arithmetic values
+  template <
+    typename SrcT  = SourceT,
+    typename TargT = TargetT,
+    typename InputIterator,
+    typename OutputIterator,
+    typename std::enable_if<cudf::is_timestamp_t<SrcT>::value && std::is_arithmetic<TargT>::value,
+                            void>::type* = nullptr>
+  void operator()(InputIterator begin, InputIterator end, OutputIterator out) const
+  {
+    std::transform(begin, end, out, [](auto const& e) {
+      return static_cast<TargT>(e.time_since_epoch().count());
+    });
+  }
+
+  // Convert timestamps to duration values
+  template <
+    typename SrcT  = SourceT,
+    typename TargT = TargetT,
+    typename InputIterator,
+    typename OutputIterator,
+    typename std::enable_if<cudf::is_timestamp_t<SrcT>::value && cudf::is_duration_t<TargT>::value,
+                            void>::type* = nullptr>
+  void operator()(InputIterator begin, InputIterator end, OutputIterator out) const
+  {
+    std::transform(begin, end, out, [](auto const& e) { return TargT{e.time_since_epoch()}; });
+  }
+
+  // Convert duration to arithmetic values
+  template <
+    typename SrcT  = SourceT,
+    typename TargT = TargetT,
+    typename InputIterator,
+    typename OutputIterator,
+    typename std::enable_if<cudf::is_duration_t<SrcT>::value && std::is_arithmetic<TargT>::value,
+                            void>::type* = nullptr>
+  void operator()(InputIterator begin, InputIterator end, OutputIterator out) const
+  {
+    std::transform(begin, end, out, [](auto const& e) { return static_cast<TargT>(e.count()); });
+  }
+};
+
 /**
  * @brief Creates a `device_buffer` containing the elements in the range
  * `[begin,end)`.
  *
- *
+ * @tparam ElementTo The type of element that is being created
+ * @tparam ElementFrom The type of element that is used to create elements of type `ElementTo`
  * @tparam InputIterator Iterator type for `begin` and `end`
  * @param begin Begining of the sequence of elements
  * @param end End of the sequence of elements
  * @return rmm::device_buffer Buffer containing all elements in the range
  *`[begin,end)`
  **/
-template <typename Element, typename InputIterator>
+template <typename ElementTo, typename ElementFrom, typename InputIterator>
 rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
 {
-  static_assert(cudf::is_fixed_width<Element>(), "Unexpected non-fixed width type.");
-  thrust::host_vector<Element> elements(begin, end);
-  return rmm::device_buffer{elements.data(), elements.size() * sizeof(Element)};
+  static_assert(cudf::is_fixed_width<ElementTo>(), "Unexpected non-fixed width type.");
+  cudf::size_type size = std::distance(begin, end);
+  thrust::host_vector<ElementTo> elements;
+  elements.reserve(size);
+  fixed_width_type_converter<ElementFrom, ElementTo>{}(begin, end, elements.begin());
```

Comment:
So I understand the need to convert from arithmetic types to timestamp/duration, but I don't understand the need for the converters for:
- From duration to arithmetic
- Timestamp to duration
- Timestamp to arithmetic

Reply:
i added this only to support creation of column wrapper from any fixed width type, in case a *typed* test needs the flexibility  to provide an initial set of values in any fixed width type. i'll remove them if they aren't going to be useful.

Reply:
That sounds useful for randomly generating test data...

Reply:
> That sounds useful for randomly generating test data...

I think the useful thing for generating test data is to allow arithmetic -> duration and arithmetic -> timestamp (which @sriramch already enabled). I'm less convinced about the need to go in the other direction when generating a `fixed_width_column_wrapper`.

Reply:
i have removed those conversions for now.